src/KSPACE/pppm.cpp

```diff
@@ -1944,7 +1944,8 @@ void PPPM::poisson_ik()
 
   // global energy and virial contribution
 
-  double scaleinv = 1.0/(nx_pppm*ny_pppm*nz_pppm);
+  bigint ngridtotal = (bigint) nx_pppm * ny_pppm * nz_pppm;
```

Comment:
Is it possible that nx_pppm, ny_pppm, and nz_pppm may need to also be bigint's, or do they have a limit within the bounds of 32-bit ints? If you cast those variables as bigint's in `kspace.h`, then the multiplication is implicitly type `bigint`, which also avoids the overflow.

Reply:
The grid counts per direction must be within the range of 32-bit signed ints.

Reply:
Even making the product of 2 dimensions a 64-bit integer, e.g. `bigint nxy_pppm = (bigint) nx_pppm * ny_pppm;` may be excessive, since `sqrt(2^31) = 46,341`, which is an absolutely huge grid for PPPM: 99 *trillion* grid points in 3d). But it doesn't really hurt anything.